### PR TITLE
Persist network plugin name as part of cluster network creation

### PR DESCRIPTION
**Pull request by pravisankar**
Depends on https://github.com/openshift/openshift-sdn/pull/317

Trello card: https://trello.com/c/Lr0wa4hr/228-3-detect-master-and-node-sdn-plugin-mismatch-sdn

### CODE REVIEW

**Comment by pravisankar** [comment]
@openshift/networking PTAL

**Comment by danwinship** [comment]
LGTM

File: plugins/osdn/common.go [hunk 1]
```diff
@@ -140,64 +138,72 @@ func (oc *OsdnController) validateNetworkConfig(clusterNetwork, serviceNetwork *
...
-		log.Errorf("Failed to obtain ClusterNetwork: %v", err)
-		return err
+		return fmt.Errorf("Failed to get network information: %v", err)
+	}
+	if ni.PluginName != oc.pluginName {
+		if ni.PluginName == "" {
+			// Do not return error in this case
+			log.Warningf("Either there is network plugin mismatch between OpenShift master(%q) and node(%q) or OpenShift master is running older version where we don't persist plugin name")
+		} else {
+			return fmt.Errorf("Detected network plugin mismatch between OpenShift master(%q) and node(%q)", ni.PluginName, oc.pluginName)
```
Comment (danwinship) [inline plugins/osdn/common.go]: should we (in another PR) deprecate the node-config networkPluginName argument and just get that value from the master like we do with the rest of the network config?

File: plugins/osdn/ovs/plugin.go [hunk 2]
```diff
@@ -35,10 +35,18 @@ type ovsPlugin struct {
 	multitenant bool
 }
 
-func CreatePlugin(registry *osdn.Registry, multitenant bool, hostname string, selfIP string, iptablesSyncPeriod time.Duration) (api.OsdnPlugin, error) {
-	plugin := &ovsPlugin{multitenant: multitenant}
+func isPluginMultitenant(pluginName string) bool {
+	switch strings.ToLower(pluginName) {
```
Comment (dcbw) [inline plugins/osdn/ovs/plugin.go]: how about just "return pluginName == MultiTenantPluginName" ?

**Comment by dcbw** [comment]
Needs a rebase onto the updated ClusterNetwork caching cleanup, but overall looks OK to me.

**Comment by pravisankar** [comment]
@openshift/networking updated, PTAL

**Comment by dcbw** [comment]
LGTM

**Pull request merged**